meta-resin-common/classes/image_types_resin.bbclass

```diff
@@ -175,43 +183,82 @@ IMAGE_CMD_resinos-img () {
+    fi
+
+    parted --align=none ${RESIN_RAW_IMG} mklabel ${PARTITION_TABLE_TYPE}
+
+    device_specific_configuration
 
     # resin-boot
-    START=${RESIN_IMAGE_ALIGNMENT}
+    if [ "${PARTITION_TABLE_TYPE}" = "msdos" ]; then
+        OPTS="primary fat16"
+    elif [ "${PARTITION_TABLE_TYPE}" = "gpt" ]; then
+        OPTS="resin-boot"
+    fi
+    START=$(expr ${START_OFFSET} \+ ${RESIN_IMAGE_ALIGNMENT})
     END=$(expr ${START} \+ ${RESIN_BOOT_SIZE_ALIGNED})
-    parted -s ${RESIN_RAW_IMG} unit KiB mkpart primary fat16 ${START} ${END}
-    parted -s ${RESIN_RAW_IMG} set 1 boot on
+    parted -s ${RESIN_RAW_IMG} unit KiB mkpart ${OPTS} ${START} ${END}
+    RESIN_BOOT_PN=$(parted -s ${RESIN_RAW_IMG} print | tail -n 2 | tr '\n' ' ' | awk '{print $1}')
+    parted -s ${RESIN_RAW_IMG} set ${RESIN_BOOT_PN} boot on
 
     # resin-rootA
+    if [ "${PARTITION_TABLE_TYPE}" = "msdos" ]; then
+        OPTS="primary ext4"
+    elif [ "${PARTITION_TABLE_TYPE}" = "gpt" ]; then
+        OPTS="resin-rootA"
+    fi
     START=${END}
     END=$(expr ${START} \+ ${RESIN_ROOTA_SIZE_ALIGNED})
-    parted -s ${RESIN_RAW_IMG} unit KiB mkpart primary ext4 ${START} ${END}
+    parted -s ${RESIN_RAW_IMG} unit KiB mkpart ${OPTS} ${START} ${END}
 
     # resin-rootB
+    if [ "${PARTITION_TABLE_TYPE}" = "msdos" ]; then
+        OPTS="primary ext4"
+    elif [ "${PARTITION_TABLE_TYPE}" = "gpt" ]; then
+        OPTS="resin-rootB"
+    fi
     START=${END}
     END=$(expr ${START} \+ ${RESIN_ROOTB_SIZE_ALIGNED})
-    parted -s ${RESIN_RAW_IMG} unit KiB mkpart primary ext4 ${START} ${END}
+    parted -s ${RESIN_RAW_IMG} unit KiB mkpart ${OPTS} ${START} ${END}
 
     # extended partition
     START=${END}
-    parted -s ${RESIN_RAW_IMG} -- unit KiB mkpart extended ${START} -1s
+    END=$(expr ${START} \+ ${RESIN_IMAGE_ALIGNMENT})
+    if [ "${PARTITION_TABLE_TYPE}" = "msdos" ]; then
+        parted -s ${RESIN_RAW_IMG} -- unit KiB mkpart extended ${START} -1s
+    fi
 
     # resin-state
-    START=$(expr ${START} \+ ${RESIN_IMAGE_ALIGNMENT})
+    if [ "${PARTITION_TABLE_TYPE}" = "msdos" ]; then
+        OPTS="logical ext4"
+    elif [ "${PARTITION_TABLE_TYPE}" = "gpt" ]; then
+        OPTS="resin-state"
+    fi
+    START=${END}
     END=$(expr ${START} \+ ${RESIN_STATE_SIZE_ALIGNED})
-    parted -s ${RESIN_RAW_IMG} unit KiB mkpart logical ext4 ${START} ${END}
+    parted -s ${RESIN_RAW_IMG} unit KiB mkpart ${OPTS} ${START} ${END}
+    RESIN_STATE_PN=$(parted -s ${RESIN_RAW_IMG} print | tail -n 2 | tr '\n' ' ' | awk '{print $1}')
 
     # resin-data
+    if [ "${PARTITION_TABLE_TYPE}" = "msdos" ]; then
+        OPTS="logical"
+    elif [ "${PARTITION_TABLE_TYPE}" = "gpt" ]; then
+        OPTS="resin-data"
+    fi
     START=$(expr ${END} \+ ${RESIN_IMAGE_ALIGNMENT})
-    parted -s ${RESIN_RAW_IMG} -- unit KiB mkpart logical ext4 ${START} -1s
+    parted -s ${RESIN_RAW_IMG} -- unit KiB mkpart ${OPTS} ext4 ${START} 100%
```

Comment:
We pass ext4 here for msdos and gpt but not above (we only pass it for msdos above).

Reply:
Will fix